### PR TITLE
script: update configgen to use infrastructure data json

### PR DESCRIPTION
**Pull request by williambanfield**
uses the work in https://github.com/tendermint/tendermint/pull/9592 to generate the testnet configuration files.

### CODE REVIEW

File: script/configgen.sh [hunk 1]
```diff
@@ -16,34 +16,62 @@ if [[ $(uname) == "Darwin" ]]; then
...
-seedsSlashSeparated=`echo $SEED_IPS | sed 's/,/\\\|/g'`
-for fname in `find . -path './testnet/seed*' -type f -name config.toml`; do
-	persistentPeers=`sed -rn 's/persistent_peers = "(.*)"/\1/p' $fname \
-		| tr , '\n' \
-		| grep "\($seedsSlashSeparated\)" || true`
-
-	result=`echo "$persistentPeers" | paste -s -d, -`
-	replace_str="s/persistent_peers = .*/persistent_peers = \\\"$result\\\"/g"
-	sed $INPLACE_SED_FLAG "$replace_str" $fname
-done
```
Comment (williambanfield) [inline script/configgen.sh:46]: Removed, the larger testnets no longer use `persistent_peers`, they've been updated to use the `seeds =` configuration parameter instead.

File: Makefile [hunk 2]
```diff
@@ -31,8 +31,7 @@ hosts:
 
 .PHONY: configgen
 configgen:
-	./script/configgen.sh $(VERSION_TAG) `ansible -i ./ansible/hosts --list-hosts validators | tail +2 | sed  's/ //g' | paste -s -d, -` \
-		`grep seed ./ansible/hosts | cut -d' ' -f1| paste -s -d, -`
+	./script/configgen.sh $(VERSION_TAG) ./ansible/hosts
```
Comment (sergio-mena) [inline Makefile:34]: No more sed kungfu :-)

File: script/configgen.sh [hunk 3]
```diff
@@ -16,34 +16,62 @@ if [[ $(uname) == "Darwin" ]]; then
...
+	cat <<EOF >> $OUT_PATH
+	  }
+	}
+EOF
+}
+
 VERSION=$1
-NEW_IPS=$2
-SEED_IPS=$3
+HOST_PATH=$2
```
Comment (sergio-mena) [inline script/configgen.sh:62]: HOSTS_PATH ?

File: script/configgen.sh [hunk 4]
```diff
@@ -16,34 +16,62 @@ if [[ $(uname) == "Darwin" ]]; then
...
+	OUT_PATH=$2
+	
+	cat <<EOF > $OUT_PATH
+	{
+	  "provider": "digital-ocean",
+	  "network": "0.0.0.0/0",
+	  "instances": {
+EOF
+	
+	lines=`grep '^.*name=.*$' $HOST_PATH | tr " " : `;
```
Comment (sergio-mena) [inline script/configgen.sh:32]: [for my own instruction] I don't get the semicolon

File: script/configgen.sh [hunk 5]
```diff
@@ -16,34 +16,62 @@ if [[ $(uname) == "Darwin" ]]; then
...
+EOF
+		if [ $i -lt $count ]; then
+	cat <<EOF >> $OUT_PATH
+	    },
+EOF
+		else
+	cat <<EOF >> $OUT_PATH
+	    }
+EOF
+		fi
```
Comment (sergio-mena) [inline script/configgen.sh:51]: Could you align these this line 53 for better readability?
(not the "EOF" lines)

**Review by sergio-mena — APPROVED** [review]
You got rid of the ugliest parts of the configgen script... well, except the MacOS-induced ugliness at the beginning :-D

File: script/configgen.sh (hunk 4 above)
Comment (williambanfield) [reply script/configgen.sh:32]: Ah, not needed. Leftover from a previous way of doing this.

**Pull request merged**